### PR TITLE
Add UUID.fromString() check to AlwaysThrows

### LINKED ISSUE #3006: Add UUID check to AlwaysThrows?
It looks like the AlwaysThrows check could check UUID.fromString() too.

### PR DESCRIPTION
**Pull request by sixcorners**
Fixes #3006

### CODE REVIEW

File: core/src/test/java/com/google/errorprone/bugpatterns/AlwaysThrowsTest.java [hunk 1]
```diff
@@ -42,6 +42,21 @@ public void positive() {
...
 
+  @Test
+  public void positiveUUID() {
+    testHelper
+        .addSourceLines(
+            "Test.java",
+            "import java.util.UUID;",
+            "class T { ",
+            "  void f() {",
+            "    // BUG: Diagnostic contains: will fail at runtime with a IllegalArgumentException",
```
Comment (sixcorners) [inline core/src/test/java/com/google/errorprone/bugpatterns/AlwaysThrowsTest.java:53]: should this be "an IllegalArgumentException"?

**Comment by kluever** [comment]
I'm confused...

`UUID.fromString(...)` only [throws](https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html#fromString-java.lang.String-) "...If name does not conform to the string representation as described in [toString()](https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html#toString--)"

**Review by cushon — APPROVED** [review]

**Comment by cushon** [comment]
I think this addition LGTM

> I'm confused...
> 
> `UUID.fromString(...)` only [throws](https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html#fromString-java.lang.String-) "...If name does not conform to the string representation as described in [toString()](https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html#toString--)"

Maybe the check could be better named, go/AlwaysThrows reports calls of APIs where the arguments are available at compile-time, and we can tell that the call will throw, e.g. it would report an error for `UUID.fromString("nope")`.

File: core/src/test/java/com/google/errorprone/bugpatterns/AlwaysThrowsTest.java (hunk 1 above)
Comment (cushon) [reply core/src/test/java/com/google/errorprone/bugpatterns/AlwaysThrowsTest.java:53]: I suppose it should, we could consider updating [the logic that constructs the diagnostic](https://github.com/google/error-prone/pull/3012/files#diff-870a751c3cdcc95fa86d5c4bcceba95518f3d400c7752641785735b03a74bbb1R189) to check for vowels as a follow-up

**Pull request closed without merging**